### ISSUE TITLE
Fix findAll (XML serialization) function in GenericController/GenericServiceBasedControllerImpl

### ISSUE THREAD

**Issue by jvillanti** [resthub/resthub-spring-stack#74]
After refactoring, the function "findAll" in GenericController/GenericServiceBasedControllerImpl mapped onto "/" (paged response of entities) is no longer working (involved : org.resthub.web.response.PageResponse)

Stacktrace : 

```
[com.sun.istack.internal.SAXException2: class org.resthub.toto.model.Sample nor any of its super class is known to this context.
javax.xml.bind.JAXBException: class org.resthub.toto.model.Sample nor any of its super class is known to this context.]
```

**Comment by sdeleuze** [comment]
We now use Jackson XML serialization, but XML version of findAll is broken.
The related bug that should be fixed is https://github.com/FasterXML/jackson-dataformat-xml/issues/6

**Comment by sdeleuze** [comment]
Waiting for Jackson 2.1, it will fix this issue

**Comment by sdeleuze** [comment]
Jackson 2.1 will not support root non wrapped values, so we should use plain arrays instead of collections for the moment

**Comment by houssemba** [comment]
Problem fixed by upgrading to the last version. (fixed on Jackson 2.3.0)

**Comment by bmeurant** [comment]
Good news ! I will soon check and release some related tests in order to close this issue.

Thank you for raising this point.
